spec.bs

```diff
@@ -2843,17 +2843,135 @@ a [=list=] of [=interest groups=] |bidIgs|, and a [=list=] of [=bid debug report
+   whose [=interest group/owner=] is |response|'s [=server auction response/interest group owner=] and
+   [=interest group/name=] is |response|'s [=server auction response/interest group name=]. Return failure if none found.
+1. If |winningGroup|'s [=interest group/ads=] is null, then return failure.
+1. Let |winningAd| be null.
+1. [=list/For each=] |ad| of |winningGroup|'s [=interest group/ads=]:
+  1. If |response|'s [=server auction response/ad render url=] is |ad|'s  [=interest group ad/render url=], then set
+    |winningAd| to |ad|, and [=iteration/break=].
+1. If |ad| is null, then return failure.
+1. Let |winningAdComponents| be a new [=list=] of [=interest group ads=].
+1. [=list/For each=] |componentAd| in |response|'s [=server auction response/ad components=]:
+  1. Let |ad| be the [=interest group ad=] from |winningGroup|'s
+     [=interest group/ad components=] where the [=interest group ad/render url=]
+     equals |componentAd|. If there is no matching element, return failure.
+  1. [=list/Append=] |ad| to |winningAdComponents|.
+1. Let |winningBid| be a new [=generated bid=] with the following [=struct/items=]:
+   : [=generated bid/id=]
+   :: (0, 0)
+   : [=generated bid/for k-anon auction=]
+   :: TODO
+   : [=generated bid/bid=]
+   :: A new [=bid with currency=] with the following [=struct/items=]:
+      : [=bid with currency/value=]
+      :: |response|'s [=server auction response/bid=]
+      : [=bid with currency/currency=]
+      :: |response|'s [=server auction response/bid currency=]
+
+  : [=generated bid/bid in seller currency=]
+  :: TODO
+  : [=generated bid/ad=]
+  :: |response|'s [=server auction response/ad render url=]
+  : [=generated bid/ad descriptor=]
+  :: |winningAd|
+  : [=generated bid/ad component descriptors=]
+  :: |winningAdComponents|
+  : [=generated bid/ad cost=]
+  :: Null
+  : [=generated bid/modeling signals=]
+  :: Null
+  : [=generated bid/interest group=]
+  :: |winningGroup|
+  : [=generated bid/bid ad=]
+  :: |winningAd|
+  : [=generated bid/modified bid=]
+  :: Null
+  : [=generated bid/bid duration=]
+  :: 0
+  : [=generated bid/component seller=]
+  :: Null
+  : [=generated bid/number of mandatory ad components=]
+  :: [=list/Size=] of |winningAdComponents|
+1. Let |buyerReportingResult| be a new [=reporting result=] with the following [=struct/items=]:
+   : [=reporting result/report url=]
+   :: The value of |response|'s [=server auction response/buyer reporting=]'s [=server auction reporting info/reporting url=].
+   : [=reporting result/reporting beacon map=]
+   :: The value of |response|'s [=server auction response/buyer reporting=]'s [=server auction reporting info/beacon urls=].
+
+1. Let |sellerReportingResult| be a new [=reporting result=] with the following [=struct/items=]:
+   : [=reporting result/report url=]
+   :: The value of |response|'s [=server auction response/top level seller reporting=]'s [=server auction reporting info/reporting url=].

```

Comment:
just to double check, this reporting url is not set in the spec, which I would expect to be set in the PR that parses the server response, right?

Reply:
I meant where would we get the `|response|'s [=server auction response/top level seller reporting=]'s [=server auction reporting info/reporting url=]` from, since it's not set in our spec (also your PR). IETF spec is fine, which parses the cbor message, and converts strings to URLs stuff. Just wanted to make sure it'll be somewhere.